filesystem/path.h

```diff
@@ -18,6 +18,7 @@
 #include <cstdlib>
 #include <cerrno>
 #include <cstring>
+#include <climits>
```

Comment:
Without it I get on cygwin:

filesystem/path.h:83:19: error: ‘PATH_MAX’ was not declared in this scope
         char temp[PATH_MAX];